### PR TITLE
ENH: Port GIFTI/CIFTI interfaces from fMRIPrep

### PR DESCRIPTION
**Pull request by mgxd**
Bring in interfaces introduced in https://github.com/nipreps/fmriprep/pull/3011

### CODE REVIEW

File: niworkflows/interfaces/surf.py [hunk 1]
```diff
@@ -558,6 +558,61 @@ def _run_interface(self, runtime):
...
+    )
+    thickness_file = File(exists=True, mandatory=True, desc='input GIFTI file')
+
+
+class CreateSurfaceROIOutputSpec(TraitedSpec):
+    roi_file = File(desc='output GIFTI file')
+
+
+class CreateSurfaceROI(SimpleInterface):
+    """Prepare GIFTI shape file for use in"""
```
Comment (effigies) [inline niworkflows/interfaces/surf.py:577]: Looks like I failed to comment this one properly.

```suggestion
    """Prepare GIFTI shape file for use in cortical masking

    Distilled from the FreeSurfer2CaretConvertAndRegisterNonlinear.sh script in
    DCAN-HCP PostFreeSurfer scripts (as of commit 9291324). The relevant lines
    are 277-290.
    """
```

File: niworkflows/interfaces/surf.py [hunk 2]
```diff
@@ -558,6 +558,61 @@ def _run_interface(self, runtime):
...
+
+    input_spec = CreateSurfaceROIInputSpec
+    output_spec = CreateSurfaceROIOutputSpec
+
+    def _run_interface(self, runtime):
+        subject, hemi = self.inputs.subject_id, self.inputs.hemisphere
+        if not isdefined(subject):
+            subject = 'sub-XYZ'
+        img = nb.GiftiImage.from_filename(self.inputs.thickness_file)
+        # wb_command -set-structure
```
Comment (effigies) [inline niworkflows/interfaces/surf.py:587]:
```suggestion
        # wb_command -set-structure (L282)
```

File: niworkflows/interfaces/surf.py [hunk 3]
```diff
@@ -558,6 +558,61 @@ def _run_interface(self, runtime):
...
+        img.meta["AnatomicalStructurePrimary"] = {'L': 'CortexLeft', 'R': 'CortexRight'}[hemi]
+        darray = img.darrays[0]
+        # wb_command -set-map-names
```
Comment (effigies) [inline niworkflows/interfaces/surf.py:590]:
```suggestion
        # wb_command -set-map-names (L284)
```

File: niworkflows/interfaces/surf.py [hunk 4]
```diff
@@ -558,6 +558,61 @@ def _run_interface(self, runtime):
...
+        meta = darray.meta
+        meta['Name'] = f"{subject}_{hemi}_ROI"
+        # wb_command -metric-palette calls have no effect on ROI files
```
Comment (effigies) [inline niworkflows/interfaces/surf.py:593]:
```suggestion
        # wb_command -metric-palette calls (L285, L289) have no effect on ROI files
```

File: niworkflows/interfaces/surf.py [hunk 5]
```diff
@@ -558,6 +558,61 @@ def _run_interface(self, runtime):
...
+
+        # Compiling an odd sequence of math operations that works out to:
```
Comment (effigies) [inline niworkflows/interfaces/surf.py:595]:
```suggestion
        # Compiling an odd sequence of math operations (L283, L288, L290) that work out to:
```

**Review by effigies — COMMENTED** [review]
Honestly skimmed over the workbench; given that you gave it a thorough read-through after I wrote it, I'm comfortable with it as long as you just copied. If you made any changes you want reviewed, could you highlight them?

Made some suggestions to improve the documentation in the Python GIFTI interface.

**Review by effigies — APPROVED** [review]

**Comment by mgxd** [comment]
Nope, no changes

**Comment by effigies** [comment]
Cool. Since all changes just merged are in docstrings or comments, I will merge without wasting CI cycles.

**Pull request merged**